pkg/ccl/multiregionccl/multiregion_test.go

```diff
@@ -160,3 +161,21 @@ func TestGlobalReadsAfterEnterpriseDisabled(t *testing.T) {
 	_, err = sqlDB.Exec(`ALTER TABLE t2 CONFIGURE ZONE USING global_reads = false`)
 	require.NoError(t, err)
 }
+
+// Regression test for https://github.com/cockroachdb/cockroach/issues/83076
+func TestMutiRegionCreateUniqueExpressionIndexOnRegionalByTable(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	defer utilccl.TestingEnableEnterprise()()
+
+	_, sqlDB, cleanup := multiregionccltestutils.TestingCreateMultiRegionCluster(
+		t, 3 /* numServers */, base.TestingKnobs{},
+	)
+	defer cleanup()
+	tdb := sqlutils.MakeSQLRunner(sqlDB)
+
+	tdb.Exec(t, `CREATE DATABASE test_db PRIMARY REGION "us-east1" REGIONS "us-east2"`)
+	tdb.Exec(t, `USE test_db`)
+	tdb.Exec(t, `CREATE TABLE t(s STRING) LOCALITY REGIONAL BY ROW`)
+	tdb.Exec(t, `CREATE UNIQUE INDEX idx ON t(lower(s))`)
```

Comment:
Could this be a logic test instead? That would be preferable, if possible.

Reply:
+1, the `regional_by_row` logic tests should setup a multi-region cluster.